diff --git a/plugin/file/file.go b/plugin/file/file.go
index f50c3d09..3b69e332 100644
--- a/plugin/file/file.go
+++ b/plugin/file/file.go
@@ -96,6 +96,8 @@ func (f File) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (i
        case NoData:
        case NameError:
                m.Rcode = dns.RcodeNameError
+
+               return plugin.NextOrFailure(f.Name(), f.Next, ctx, w, r)
        case Delegation:
                m.Authoritative = false
        case ServerFailure: